### PR TITLE
Callback API for the -wasm package

### PR DESCRIPTION
**Pull request by staltz**
Pending on PR #2 being merged first.

### CODE REVIEW

File: test/test.mjs [hunk 1]
```diff
@@ -0,0 +1,170 @@
+import * as Comlink from "comlink";
+import singleAuthorMsgs from "./data/singleAuthorMsgs.js";
+import multiAuthorMsgs from "./data/multiAuthorMsgs.js";
+import validMsg from './data/valid.js';
+
+// We can't just require ../index because Karma doesn't bundle `new Worker`
```
Comment (mycognosist) [inline test/test.mjs:6]: I would not have figured this out by myself 0_o

**Review by mycognosist — APPROVED** [review]
Looks good! I'd love to chat (briefly) about mocha vs. jasmine on a call some time, just to have a better understanding of their various strengths and uses.

The cb interface is neat and I'm sure it will feel a lot more ergonomic for users of the package.

**Comment by staltz** [comment]
@mycognosist Thanks for the review here too. Mocha and Jasmine are quite similar, I just replaced them this time because I was losing my patience over Jasmine specific bugs like something related to spies, and I thought that I could just try out Mocha because I know it's been rock solid in the past for me.

What's missing is that I didn't manage to use nice assertion tools like `toBeFalsy`. I tried to import `chai` which has these assertion helpers, but the ESM complication gave me headaches so I just went for a super simple assertion method, which is an `if`/`else` that will either fail or succeed the test.

**Comment by staltz** [comment]
I'll merge this one into branch `api-browser`.

**Pull request merged**